engine/common/synchronization/engine.go

```diff
@@ -303,6 +314,7 @@ func (e *Engine) onBlockResponse(originID flow.Identifier, res *messages.BlockRe
 		}
 		e.comp.SubmitLocal(synced)
 	}
+	e.log.Debug().Uint64("min", min).Uint64("max", max).Msg("received block response")
```

Comment:
We would only print this log _after_ all blocks have been handled, which makes the log ordering a bit confusing:
```
processing block at height 100
processing block at height 101
processing block at height 102
...
processing block at height 110
received block response min: 100, max: 110
```

Usually, the first block is the min block, and the last block is the max block, we could just log with that, if someone is cheating that's not a big deal, they only mess up our Debug log.

So I suggest to just print this log at the beginning of this function.

Reply:
hmmm, ok, this makes more sense. I was under an impression that order is not really guaranteed, but even if, it doesn't matter that much